tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpAdapterBase.java

```diff
@@ -1,289 +1,151 @@
-                .setHandler(context.asyncAssertFailure(t -> {
-                    context.assertTrue(ClientErrorException.class.isInstance(t));
-                    context.assertEquals(HttpURLConnection.HTTP_BAD_REQUEST, ((ClientErrorException) t).getErrorCode());
-                    completionTracker.complete();
-                }));
-        completionTracker.awaitSuccess();
-    }
-
-    /**
-     * Verifies that the AMQP adapter rejects messages from a device that belongs to a tenant for which it has been
-     * disabled. This test case only considers events since telemetry senders do not wait for response from the adapter.
-     * 
-     * @param context The Vert.x test context.
-     */
-    @Test
-    public void testUploadMessageFailsForTenantDisabledAdapter(final TestContext context) {
-        final Async completionTracker = context.async();
-
-        final String tenantId = helper.getRandomTenantId();
-        final String deviceId = helper.getRandomDeviceId(tenantId);
-        final String username = IntegrationTestSupport.getUsername(deviceId, tenantId);
-        final String password = "secret";
-
-        final TenantObject tenant = TenantObject.from(tenantId, true);
-        tenant.addAdapterConfiguration(TenantObject.newAdapterConfig(Constants.PROTOCOL_ADAPTER_TYPE_AMQP, false));
+        final Message msg = ProtonHelper.message("msg payload");
 
-        final Async addTenantTracker = context.async();
+        msg.setContentType("application/vnd.eclipse-hono-empty-notification");
 
-        helper.registry
-                .addDeviceForTenant(tenant, deviceId, password)
-                .setHandler(context.asyncAssertSuccess(ok -> addTenantTracker.complete()));
+        sender.send(msg).setHandler(context.asyncAssertFailure(t -> {
 
-        addTenantTracker.await();
-
-        final HonoClient client = prepareAmqpAdapterClient(username, password);
-        client.connect(new ProtonClientOptions()).compose(ok -> {
-            clientsToShutdown.add(client);
-            return getOrCreateSender(tenantId, client);
-        }).compose(sender -> {
-            final Message msg = ProtonHelper.message("some payload");
-            return sender.send(msg);
-        }).setHandler(context.asyncAssertFailure(t -> {
             context.assertTrue(ClientErrorException.class.isInstance(t));
 
-            // HonoClient maps AmqpError.UNAUTHORIZED_ACCESS to HttpURLConnection.HTTP_BAD_REQUEST
-            // -> so we test the message
-
-            final String expectedErrorMsg = String
-                    .format("This adapter is not enabled for tenant [tenantId: %s].", tenantId);
-
-            context.assertEquals(expectedErrorMsg, ((ClientErrorException) t).getMessage());
+            context.assertEquals(HttpURLConnection.HTTP_BAD_REQUEST, ((ClientErrorException) t).getErrorCode());
 
             completionTracker.complete();
         }));
-        completionTracker.awaitSuccess();
+        completionTracker.await(1000);
```

Comment:
there is no need to create an `Async` and wait for it to complete. The test runner will already wait for the `context.asyncAssertFailure()` handler to fail and then perform the additional assertions you added there.

Reply:
i see, thanks